### PR TITLE
Feature/edit aatf org

### PR DESCRIPTION
**Pull request by JohnPadgettCivica**

### CODE REVIEW

File: src/EA.Weee.RequestHandlers/Organisations/OrganisationByIdHandler.cs [hunk 1]
```diff
@@ -28,17 +29,18 @@ public async Task<OrganisationData> HandleAsync(GetOrganisationInfo query)
...
 
             if (org == null)
             {
-                throw new ArgumentException(string.Format("Could not find an organisation with id {0}",
-                    query.OrganisationId));
+                throw new ArgumentException($"Could not find an organisation with id {query.OrganisationId}");
             }
 
-            return organisationMap.Map(org);
+            var organisationData = organisationMap.Map(org);
```
Comment (SamGoodeySFW) [inline src/EA.Weee.RequestHandlers/Organisations/OrganisationByIdHandler.cs]: Unit test to make sure that mapper is run with the returned organisation

File: src/EA.Weee.Core.Tests.Unit/Organisations/OrganisationDataTests.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+            var organisation = new OrganisationData()
+            {
+                OrganisationType = OrganisationType.RegisteredCompany
+            };
+
+            organisation.IsRegisteredCompany.Should().BeTrue();
+        }
+
+        [Fact]
+        public void IsRegisteredCompany_GivenNotRegisteredCompany_FalseReturned()
```
Comment (willdot) [inline src/EA.Weee.Core.Tests.Unit/Organisations/OrganisationDataTests.cs]: Can't see an assertion here?

File: src/EA.Weee.Web.Tests.Unit/Areas/Admin/Mapping/AatfDataToAatfDetailsViewModelMapTests_2.cs [hunk 3]
```diff
@@ -0,0 +1,60 @@
+﻿namespace EA.Weee.Web.Tests.Unit.Areas.Admin.Mapping
```
Comment (SamGoodeySFW) [inline src/EA.Weee.Web.Tests.Unit/Areas/Admin/Mapping/AatfDataToAatfDetailsViewModelMapTests_2.cs]: Got 2 view model tests, reminder to remove 1 of them

File: src/EA.Weee.Web/Areas/Admin/ViewModels/Organisation/EditSoleTraderOrIndividualOrganisationDetailsViewModel.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    using System.ComponentModel.DataAnnotations;
+    using EA.Weee.Core.DataStandards;
+    using EA.Weee.Core.Organisations;
+    using EA.Weee.Core.Shared;
+
+    public class EditSoleTraderOrIndividualOrganisationDetailsViewModel : EditOrganisationDetailsViewModelBase
+    {       
+        [Required]
+        [DisplayName("Business trading name")]
+        [StringLength(CommonMaxFieldLengths.DefaultString)]
```
Comment (SamGoodeySFW) [inline src/EA.Weee.Web/Areas/Admin/ViewModels/Organisation/EditSoleTraderOrIndividualOrganisationDetailsViewModel.cs]: Unit tests to ensure attributes are set

File: src/EA.Weee.RequestHandlers/Organisations/OrganisationByIdHandler.cs (hunk 1 above)
Comment (JohnPadgettCivica) [reply src/EA.Weee.RequestHandlers/Organisations/OrganisationByIdHandler.cs]: Added test

File: src/EA.Weee.Web/Areas/AatfReturn/Mappings/ToViewModel/ReturnToObligatedViewModelMap.cs [hunk 5]
```diff
@@ -76,10 +76,10 @@ public ObligatedViewModel Map(ReturnToObligatedViewModelMapTransfer source)
 
             if (source.PastedData != null)
             {
-                var obligatedPastedValues = new ObligatedPastedValues();
```
Comment (JB12298) [inline src/EA.Weee.Web/Areas/AatfReturn/Mappings/ToViewModel/ReturnToObligatedViewModelMap.cs]: Not your change, but Line 61's SiteName Null Check doesn't have a test

File: src/EA.Weee.Web.Tests.Unit/Areas/Admin/Mapping/AatfDataToAatfDetailsViewModelMapTests_2.cs (hunk 3 above)
Comment (JohnPadgettCivica) [reply src/EA.Weee.Web.Tests.Unit/Areas/Admin/Mapping/AatfDataToAatfDetailsViewModelMapTests_2.cs]: Removed

File: src/EA.Weee.Web/Areas/Admin/Controllers/AddAatfController.cs [hunk 6]
```diff
@@ -95,9 +95,10 @@ public async Task<ActionResult> SearchResults(string searchTerm)
         {
             SetBreadcrumb(InternalUserActivity.CreateAatf);
```
Comment (JB12298) [inline src/EA.Weee.Web/Areas/Admin/Controllers/AddAatfController.cs]: Can the FetchSearchResultsJson method be removed? @willdot

Comment (willdot) [reply src/EA.Weee.Web/Areas/Admin/Controllers/AddAatfController.cs]: This is for the search as your type feature, so if that's enabled, it needs to stay in :)

File: src/EA.Weee.Core.Tests.Unit/Organisations/OrganisationDataTests.cs (hunk 2 above)
Comment (JohnPadgettCivica) [reply src/EA.Weee.Core.Tests.Unit/Organisations/OrganisationDataTests.cs]: Added assert

File: src/EA.Weee.Web/Areas/Admin/Controllers/AddAatfController.cs (hunk 6 above)
Comment (JohnPadgettCivica) [reply src/EA.Weee.Web/Areas/Admin/Controllers/AddAatfController.cs]: unfortunately this is existing code and didnt have unit tests... have just moved it from the scheme controller...

File: src/EA.Weee.Web/Areas/Admin/ViewModels/Organisation/EditSoleTraderOrIndividualOrganisationDetailsViewModel.cs (hunk 4 above)
Comment (JohnPadgettCivica) [reply src/EA.Weee.Web/Areas/Admin/ViewModels/Organisation/EditSoleTraderOrIndividualOrganisationDetailsViewModel.cs]: unit tests added

File: src/EA.Weee.Web/Areas/Admin/Controllers/AddAatfController.cs (hunk 6 above)
Comment (JohnPadgettCivica) [reply src/EA.Weee.Web/Areas/Admin/Controllers/AddAatfController.cs]: The controller was taken from the external screens and looks like didn't originally have a test... will create some

File: src/EA.Weee.Web/Areas/AatfReturn/Mappings/ToViewModel/ReturnToObligatedViewModelMap.cs (hunk 5 above)
Comment (JohnPadgettCivica) [reply src/EA.Weee.Web/Areas/AatfReturn/Mappings/ToViewModel/ReturnToObligatedViewModelMap.cs]: Added unit test

File: src/EA.Weee.Web/Areas/Admin/Controllers/AddAatfController.cs (hunk 6 above)
Comment (JohnPadgettCivica) [reply src/EA.Weee.Web/Areas/Admin/Controllers/AddAatfController.cs]: Added tests

**Pull request merged**